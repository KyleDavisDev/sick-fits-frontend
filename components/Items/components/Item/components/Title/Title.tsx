import * as React from "react";
import Link from "next/link";
import { StyledText } from "./TitleStyle";

interface ITitleProps {
  children: string;
  id: string;
}

const Title: React.FunctionComponent<ITitleProps> = props => {
  return (
    <StyledText>
      <Link href={{ pathname: "/item", query: { id: props.id } }}>
        <a>{props.children}</a>
      </Link>
    </StyledText>
  );
};

export default Title;
