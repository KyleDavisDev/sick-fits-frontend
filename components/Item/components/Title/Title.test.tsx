import * as React from "react";
import { shallow } from "enzyme";
import Title from "./Title";

describe("<Title />", () => {
  it("renders", () => {
    const wrapper = shallow(<Title id={"123"}>test</Title>);

    expect(wrapper).toBeTruthy();
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<Title id={"123"}>test</Title>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders the expected title", () => {
    let title = "Here is a test title";
    const wrapper = shallow(<Title id={"123"}>{title}</Title>);

    // check title
    expect(wrapper.find("a").text()).toEqual(title);

    // update title, check again
    title = "Another title";
    wrapper.setProps({ children: title });
    expect(wrapper.find("a").text()).toEqual(title);

    // update title, check again
    title = "Lets go again!!";
    wrapper.setProps({ children: title });
    expect(wrapper.find("a").text()).toEqual(title);
  });
});
