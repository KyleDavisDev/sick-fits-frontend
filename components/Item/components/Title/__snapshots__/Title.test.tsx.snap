// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches snapshot 1`] = `
<styled.h3>
  <Link
    href={
      Object {
        "pathname": "/item/view",
        "query": Object {
          "id": "123",
        },
      }
    }
  >
    <a>
      test
    </a>
  </Link>
</styled.h3>
`;
